Remotion/Validation/UnitTests/Implementation/DelegateBasedValidationMessageTest.cs

```diff
@@ -15,29 +15,55 @@
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using System.Globalization;
 using NUnit.Framework;
+using Remotion.Validation.Implementation;
 
 namespace Remotion.Validation.UnitTests.Implementation
 {
   [TestFixture]
   public class DelegateBasedValidationMessageTest
   {
     [Test]
-    [Ignore("RM-5906")]
-    public void Format_UsesSuppliedFormatProviderInsteadOfCurrentCulture ()
+    [SetCulture("en-US")]
+    [SetUICulture("de-CH")]
+    public void Format_WithFormatProvider_UsesSuppliedFormatProviderInsteadOfCurrentCulture ()
     {
+      Func<string> validationMessageProvider = () => "{0} : {1}";
+      var validationMessage = new DelegateBasedValidationMessage(validationMessageProvider);
+      var culture = CultureInfo.GetCultureInfo("fr-FR");
+      var formatProvider = CultureInfo.GetCultureInfo("de-AT");
+      var parameters = new object[] { 3123.312, new DateTime(2023, 05, 15) };
+
+      var result = validationMessage.Format(culture, formatProvider, parameters);
+
+      Assert.That(result, Is.EqualTo("3123,312 : 15.05.2023 00:00:00"));
     }
 
     [Test]
-    [Ignore("RM-5906")]
-    public void Format_UsesSuppliedCultureInsteadOfCurrentUICulture ()
+    [SetCulture("it-IT")]
+    [SetUICulture("de-CH")]
+    public void Format_WithNullAsFormatProvider_UsesSuppliedCultureInsteadOfCurrentUICulture ()
     {
+      Func<string> validationMessageProvider = () => "{0} : {1}";
+      var validationMessage = new DelegateBasedValidationMessage(validationMessageProvider);
+      var culture = CultureInfo.GetCultureInfo("fr-FR");
+      var parameters = new object[] { 3123.312, new DateTime(2023, 05, 15) };
+
+      var result = validationMessage.Format(culture, null, parameters);
+
+      Assert.That(result, Is.EqualTo("3123.312 : 05/15/2023 00:00:00"));
     }
 
```

Comment:
I am not sure why we need 2 tests here, since we just want to check if the passed culture is set as UICulture when calling the delegate, it doesn't actually matter with or without formatprovider or have I missed something?

Reply:
oh, yeah, I may have misread the test name :/